src/pages/workspaces/workspace/notebooks/TerminalLauncher.js

```diff
@@ -101,10 +101,20 @@ const TerminalLauncher = _.flow(
       })
     } else {
       return div({ style: { padding: '2rem' } }, [
+        'Creating Stopping Starting Updating Deleting'.includes(clusterStatus) &&
         spinner({ style: { color: colors.primary(), marginRight: '0.5rem' } }),
-        (clusterStatus === 'Creating' || !cluster) ?
-          'Creating runtime environment. You can navigate away and return in 5-10 minutes.' :
-          'Starting runtime environment, this may take up to 2 minutes.'
+        Utils.switchCase(clusterStatus,
+          ['Creating', () => 'Creating notebook runtime environment. You can navigate away and return in 3-5 minutes.'],
+          ['Stopping', () => 'Notebook runtime environment is stopping, which takes ~4 minutes. You can restart it after it finishes.'],
+          ['Starting', () => 'Starting notebook runtime environment, this may take up to 2 minutes.'],
+          ['Updating', () => 'Updating notebook runtime environment. You can navigate away and return in 3-5 minutes.'],
+          ['Deleting', () => 'Deleting notebook runtime environment, you can create a new one after it finishes.'],
+          ['Stopped', () => 'Notebook runtime environment is stopped. Start it to edit your notebook or use the terminal.'],
+          ['Deleted', () => 'No existing notebook runtime environment, you can create a new one to edit your notebook or use the terminal.'],
+          ['Error', () => 'Error with the notebook runtime environment, please try again.'],
+          ['Unknown', () => 'Error with the notebook runtime environment, please try again.'],
+          [Utils.DEFAULT, () => 'Invalid notebook runtime environment state, please try again.']
```

Comment:
Hi, I was looking at a user-liaison question and noticed this PR. A user was saying they keep getting this `DEFAULT` error message when launching a terminal and have to manually refresh the page. When I try to launch a terminal I briefly see this message then the page seems to refresh by itself. Any idea what the issue might be for the user?

Reply:
Looks like this default case is catching the condition where we don't yet know the cluster status (the ajax call hasn't returned yet) or there is no cluster. We should clean that up. That said, I expect the behavior you experienced (brief flash of this message), not requiring a manual refresh. Is there a difference between the cases of having a running cluster, a paused cluster, and no cluster?

Reply:
Hm, I think I see the expected behavior in all cases: a brief flash of this message followed by an action (e.g. cluster creation) and the appropriate message. I'm not sure why the user was seeing this.